src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
I just meant for the proposed sampling method above it will always split the 10 sampled points evenly regardless of similarity metric (obviously not the full set of points if you are only sampling 10). 

Yep, completely agree. Not fully sure on the theoretical implications of the sampling method but seems like a nice balance of being smarter about the hyperplane split (splitting k points instead of just 2) but also maintaining nondeterminism so as to avoid trees replicating splits.